app/database.py

```diff
@@ -23,83 +25,142 @@ def get_database():
+    return str(json.dumps(({'classes': classes})))
 
-# returns any column passed in as an argument.
-# For Example: 'flask fetch_database course_id'
-# will return all of the course ids in the database
-@click.command('fetch_database')
-@click.argument('field')
-@with_appcontext
-def fetch_database_command(field):
-    fetch_database(field)
 
-# fills the database with a file from the given path.
-# Will throw an error if the file passed is not in
-# the correct format.
-@click.command('fill_database')
+def get_profs():
+    db = get_db()
+    cursor = db.cursor()
+    statement = 'SELECT DISTINCT professor FROM winter2017'
+    rows = cursor.execute(statement).fetchall()
+    lst = []
+    for row in rows:
+        lst.append(row[0])
+    return lst
+
+
+def get_subject():
+    db = get_db()
+    cursor = db.cursor()
+    statement = 'SELECT DISTINCT (SUBSTR(title, 0, 5)) FROM winter2017'
+    rows = cursor.execute(statement).fetchall()
+    lst = []
+    for row in rows:
+        lst.append(row[0])
+    return lst
+
+
+'''
+The following methods are for command line manipulation of the this app.  They are not necessary when working within
+the webpage.  
+
+fills the database with a file from the given path.Will throw an error if the file passed is not in the correct format.
+'''
+
+
+@click.command('fill_db')
 @click.argument('file')
+#@click.argument('semester')
 @with_appcontext
-def fill_database_command(file):
-    from . import parser
-    parser.parse(file)
-    event = 'Database filled'
+def fill_db_command(file):
+    fill_db(file)
+    event = 'Database filled with ' + file
     log.add_event(event)
+    click.echo(event)
+
 
-# drops the table from the database, removing everything
-# in it.
-@click.command('destroy_database')
+'''
+selects all classes of a queried category.  Can be refined down to year. 
+call 'Flask show_classes <semester> <course type>'
+eg1 Flask show_classes fall2017 CPSC displays all CPSC courses
+eg2 Flask show_classes winter2017 CPSC2 displays all 2nd year courses
+'''
+
+
+@click.command('get_classes')
+@click.argument('semester')
 @with_appcontext
-def destroy_database_command():
-    destroy_database()
-    event = 'Database destroyed'
+def get_classes_command():
+    event = 'retrieve classes'
     log.add_event(event)
     click.echo(event)
+    get_events()
 
 
+'''
+These are all commands that can be used in the virtual environment.  Call them before calling flask run.
+'''
+
 
-# these are all commands that can be used in the virtual
-# environment.  Call them before calling flask run.
 def init_app(app):
-    app.teardown_appcontext(close_database)
-    app.cli.add_command(init_database_command)
-    app.cli.add_command(fill_database_command)
-    app.cli.add_command(destroy_database_command)
-    app.cli.add_command(fetch_database_command)
+    app.teardown_appcontext(close_db)
+    app.cli.add_command(fill_db_command)
+    app.cli.add_command(get_classes_command)
```

Comment:
this line can be removed along with that method as this just initializes it and would not be needed if the method is removed

Reply:
'This line' being 166